src/applications/model/gaming-streaming-server.h

```diff
@@ -87,38 +84,74 @@ class GamingStreamingServer : public Application
   uint32_t GetTotSentPackets (void);
 
   /**
-   * Erase the statistics of sent packets
+   * \brief return the number of total failed packets
+   * \return total number of failed packets
+   */
+  uint32_t GetTotFailedPackets (void);
+
+  /**
+   * \brief return the total bytes sent
+   * \return the total bytes sent
+   */
+  uint32_t GetTotSentBytes (void);
+
+  /**
+   * \brief Erase the statistics of sent packets
    */
   void EraseStatistics (void);
 
+  /**
+   * \brief Add parameters of new stream
+   *
+   * \param packetSize A random number generator for packet size
+   * \param interArrivalTime A random number generator for inter-arrival time
+   */
+  void AddNewTraffic (Ptr<RandomVariableStream> packetSize, Ptr<RandomVariableStream> interArrivalTime);
+
+  virtual void StartApplication (void);
+  virtual void StopApplication (void);
+
 protected:
   /**
-   * \brief Generate and send packets based on the different distribution
+   * \brief Generate and send packets based on the different random
+   * distribution for packet sizes and inter-arrival times
    *
-   * \param pktDist random variable which defines the size of generated packets
-   * \param iatDist random variable which defines inter-arrival times between the packets
+   * \param idx index of stream within the streams vector
    */
-  void SendPacket (Ptr<RandomVariableStream> pktDist, Ptr<RandomVariableStream> iatDist);
+  void Send (int idx);
```

Comment:
I have a problem when using the `TrafficStream` struct.
 Here, in the `ScheduleNextTx` method, I schedule the next `sendEvent` by passing the `traffic` (an instant of TrafficStream) to the `Simulator::Schedule` and in the same time I save this event in the `sendEvent` element of `traffic` like:

`traffic.sendEvent = Simulator::Schedule (nextTx, &GamingStreamingServer::Send, this, traffic);`

Everything seems fine. But I can't stop the application anymore. 
I'm not sure the exact source of the problem.

Reply:
you're probably passing the object `traffic` by value instead of reference, so you're modifying a local copy that is destroyed as soon as you exit the `ScheduleNextTx` method

Reply:
Yes, you right. 
The problem is `Simulator::Schedule` takes the functions' arguments as passed by value.

Reply:
no, i'm talking about the `TrafficStream` object: it should be passed by reference in order to modify its `sendEvent`

Reply:
Yeah, I know.
I've already fixed it.